testing/fuchsia/run_tests.py

```diff
@@ -128,9 +109,45 @@ def bundled_test_runner_of(target_id: str) -> BundledTestRunner:
       resolved_packages.add(new_package)
     else:
       resolved_packages.add(os.path.join(OUT_DIR, package))
+  return resolved_packages
+
+
+def build_test_cases(tests: dict) -> List[TestCase]:
+  test_cases = []
+  for test in [t['test_command'] for t in tests]:
+    assert test.startswith('test run ')
+    test = test[len('test run '):]
+    if ' -- ' in test:
+      index = test.index(' -- ')
+      test_cases.append(
+          TestCase(package=test[:index], args=test[index + len(' -- '):])
+      )
+    else:
+      test_cases.append(TestCase(package=test))
+  return test_cases
+
+
+def bundled_test_runner_of(target_id: str) -> BundledTestRunner:
+  log_dir = os.environ.get('FLUTTER_LOGS_DIR', '/tmp/log')
+  with open(os.path.join(os.path.dirname(__file__), 'test_suites.yaml'),
+            'r') as file:
+    tests = yaml.safe_load(file)
+  # TODO(zijiehe-google-com): Run tests with dart aot,
+  # https://github.com/flutter/flutter/issues/140179.
+  tests = list(
+      filter(
+          lambda test: 'run_with_dart_aot' not in test or test[
+              'run_with_dart_aot'] != 'true', tests
+      )
+  )
+  tests = list(
+      filter(
+          lambda test: 'variant' not in test or VARIANT == test['variant'],
+          tests
+      )
+  )
```

Comment:
These should be comprehensions rather than filter, and can also be generators rather than lists.
```suggestion

def dart_jit(test) -> bool:
    return 'run_with_dart_aot' not in test or test[
              'run_with_dart_aot'] != 'true'
              
 def foo(test) -> bool:
    return 'variant' not in test or VARIANT == test['variant']
          
  tests = (t for t in tests if dart_jit(t) and foo(t))    
```

and typings in callers can accept `Iterator[Mapping[str, list]`. (I think that's the correct type for the YAML?)

Reply:
Oh, I thought these filters will eventually be all removed one-by-one.

Done.